packages/twenty-front/src/modules/ui/input/components/internal/date/utils/getHighlightedDates.ts

```diff
@@ -0,0 +1,18 @@
+import { addDays, addMonths, startOfDay, subMonths } from 'date-fns';
+
+export const getHighlightedDates = (start: Date, end: Date): Date[] => {
+  const highlightedDates: Date[] = [];
+  const currentDate = startOfDay(new Date());
+  const twoMonthsAgo = subMonths(currentDate, 2);
+  const twoMonthsFromNow = addMonths(currentDate, 2);
+
+  let dateToHighlight = start < twoMonthsAgo ? twoMonthsAgo : start;
+  const lastDate = end > twoMonthsFromNow ? twoMonthsFromNow : end;
+
+  while (dateToHighlight <= lastDate) {
+    highlightedDates.push(dateToHighlight);
+    dateToHighlight = addDays(dateToHighlight, 1);
+  }
```

Comment:
We never generate large ranges because the disabled calendar only displays the current month and a little more. The function limits the range to two months max.